### PR TITLE
Fix typos [nfc]

### PR DESCRIPTION
**Pull request by LilithHafner**

### CODE REVIEW

**Comment by DilumAluthge** [comment]
Nice!

File: src/julia_internal.h [hunk 1]
```diff
@@ -211,13 +211,13 @@ void jl_unlock_stackwalk(int lockret) JL_NOTSAFEPOINT JL_NOTSAFEPOINT_LEAVE;
...
 #if defined(_CPU_X86_64_)
-    // This is nopl 0(%rax, %rax, 1), but assembler are incosistent about whether
+    // This is nopl 0(%rax, %rax, 1), but assembler are inconsistent about whether
     // they emit that as a 4 or 5 byte sequence and we need to be guaranteed to use
     // the 5 byte one.
 #define NOP5_OVERRIDE_NOP ".byte 0x0f, 0x1f, 0x44, 0x00, 0x00\n\t"
     uint64_t low, high;
     // This instruction sequence is promised by rr to be patchable. rr can usually
-    // also patch `rdtsc` in regular code, but without the preceeding nop, there could
+    // also patch `rdtsc` in regular code, but without the proceeding nop, there could
```
Comment (DilumAluthge) [inline src/julia_internal.h:220]: I think this should be `preceding`?

Comment (LilithHafner) [reply src/julia_internal.h:220]:
```suggestion
    // also patch `rdtsc` in regular code, but without the preceding nop, there could
```

File: doc/src/devdocs/pkgimg.md [hunk 2]
```diff
@@ -33,7 +33,7 @@ Dynamic libraries on macOS need to link against `-lSystem`. On recent macOS vers
 To that effect we link with `-undefined dynamic_lookup`.
 
 ## [Package images optimized for multiple microarchitectures](@id pkgimgs-multi-versioning)
-Similar to [multi-versioning](@ref sysimg-multi-versioning) for system images, package images support multi-versioning. If you are in a heterogenous environment, with a unified cache,
+Similar to [multi-versioning](@ref sysimg-multi-versioning) for system images, package images support multi-versioning. If you are in a heterogeneous environment, with a unified cache,
```
Comment (DilumAluthge) [inline doc/src/devdocs/pkgimg.md:36]: It looks like both [`heterogenous`](https://www.merriam-webster.com/dictionary/heterogenous) and [`heterogeneous`](https://www.merriam-webster.com/dictionary/heterogeneous) are valid spellings.

Would you happen to remember which tool (`typos` or `codespell`) flagged `heterogenous` as a misspelling?

Comment (ViralBShah) [reply doc/src/devdocs/pkgimg.md:36]: Pretty sure that heterogeneous is the more common spelling.

Comment (LilithHafner) [reply doc/src/devdocs/pkgimg.md:36]: `codespell`

**Comment by LilithHafner** [comment]
Force-push was undoing an accidental push that was meant for #51704

**Comment by nsajko** [comment]
Could you also fix the "exhasutive" on the front page of the documentation:

https://github.com/JuliaLang/julia/blob/865007d701341526bc5442fcb2409a82a82a7ccd/doc/src/index.md?plain=1#L37

**Comment by LilithHafner** [comment]
I'm going to merge this soon unless someone objects. I don't want to deal with merge conflicts and this touches 70 files.

**Comment by DilumAluthge** [comment]
I'll go through and do one last full pass now.

File: doc/src/devdocs/pkgimg.md (hunk 2 above)
Comment (DilumAluthge) [reply doc/src/devdocs/pkgimg.md:36]: I think it's fine if we standardize on the `heterogeneous` spelling, as long as we're consistent across the codebase.

File: stdlib/REPL/test/replcompletions.jl [hunk 3]
```diff
@@ -1980,7 +1980,7 @@ let (c, r, res) = test_complete_context("getkeyelem(mutable_const_prop).value.")
 end
 
 # JuliaLang/julia/#51548
-# don't return wrong result due to mutable inconsistentcy
+# don't return wrong result due to mutable inconsistency
```
Comment (DilumAluthge) [inline stdlib/REPL/test/replcompletions.jl:1983]: @simeonschaub @aviatesk 

Does this refer to the `:consistent` effect? If so, should this say "don't return wrong result due to mutable in-`:consistent`-cy", or something like that?

**Review by DilumAluthge — APPROVED** [review]
LGTM.

If we do end up needing to fix anything regarding the `inconsistentcy => inconsistency` change (`stdlib/REPL/test/replcompletions.jl` line 1983), we can do so in a follow-up PR.

**Comment by DilumAluthge** [comment]
@LilithHafner Can you add `exhasutive => exhaustive` to the list of new typos that you've reported to `typos`?

**Comment by LilithHafner** [comment]
Already have: https://github.com/crate-ci/typos/issues/843#issuecomment-1763053064

File: stdlib/REPL/test/replcompletions.jl (hunk 3 above)
Comment (simeonschaub) [reply stdlib/REPL/test/replcompletions.jl:1983]: It does refer to the effect, but IMHO it's fine to refer to the effect like this. Spelling it `consistent`-cy always reads a bit clunky to me

**Pull request merged**